test/tools/cli/IonProcess.java

```diff
@@ -544,14 +502,28 @@ private static Event ionStreamToEvent(IonReader ionReader) {
                     event.setDepth(ionReader.intValue());
                     break;
             }
-        }
 
+        }
+        //compare text value and binary value
+        if (textValue != null && binaryValue != null) {
+            IonSystem ionSystem = IonSystemBuilder.standard().build();
+            IonValue text = ionSystem.singleValue(textValue);
+            IonValue binary = ionSystem.singleValue(binaryValue);
+            if (!Equivalence.ionEquals(text, binary)) {
+                throw new IonException("invalid Event: Text value and Binary value are different");
+            } else {
+                IonValue v = IonSystemBuilder.standard().build().singleValue(textValue);
```

Comment:
No need to re-read it here; you've already got this value stored in `text` above.

Reply:
Changed.